### DISCUSSION TITLE
One Polefigure with contour lines for several ebsd datasets?

### DISCUSSION THREAD

**Discussion by Debo-rah** [mtex-toolbox/mtex#579] [Ask Anything] [answered]
Hi,

I made a few ebsd maps and want to combine all corresponding polefigures in one polefigure. I manage to do this whith displaying dots. 

``
plotPDF()
hold on
plotPDF
``

However, when I add 'contourf' to the plots it overwrites the figures and only displays the last one. I guess I first have to combine all ODFs to one and then plot this new ODF? 

I would really like to have one contour plot for all datasets.

**Comment by ralfHielscher** [comment]
Yeah, you need to compute a combined ODF. Stacking contours makes not so much sense.

Ralf.

**Reply by Debo-rah** [reply]
Hi Ralf,
yes, thats what I guessed, but how can I combine the ODFs?
I only found the function plus(odf1,odf2), but my ebsd datasets do not have the same size and i also want to add more than 2.

Thanks for helping,
Deborah

**Comment by ralfHielscher** [comment, marked as answer]
You can do simply
```matlab
odf1 + odf2 + odf3 
```
I might also make sense to weight the odfs according to the size of the EBSD maps.
you can also compute the ODF directly from all maps by
```matlab
odf = calcODF([ebsd1.orientations;ebsd2.orientations;ebsd3.orientations])
```
Ralf.

**Reply by Debo-rah** [reply]
Thanks a lot, that worked very well!

**Comment by zmichels** [comment]
Hi Deborah,

I think first you want to combine all the orientations that you’ve used for constructing each ODF, and then calculate an ODF from the combined set of orientations.

For example... if you have two sets of orientations, o1 and o2 that you used to compute ODF1 and ODF2... instead combine/concatenate o1 & o2:

o = [o1, o2]

Then compute an ODF from o.

Cheers,
Zach

**Reply by Debo-rah** [reply]
Thanks for helping Zach. I tried your solution as well, but this seems to  work only in the case that your orientation datasets have the same size. Since my maps have different sizes, I got an errormessage. 
Thanks anyways,
Deborah

**Comment by zmichels** [comment]
Deborah,

I see, yes. I think you would need to concatenate with semicolon instead (my mistake) as:

o = [o1;o2]

But it sounds like adding the weighted ODFs is fine for your need.

Cheers,
Zach
